File: frontend/src/types/call.ts
export interface CallTurn {
  timestamp: string;
  customer?: string;
  agent?: string;
}

export interface ToolCallEvent {
  timestamp: string;
  tool: string;
  arguments: Record<string, any>;
  result: Record<string, any>;
}

export interface CallRecord {
  id: string;
  call_sid: string;
  caller: string;
  recipient: string;
  status: 'active' | 'completed' | 'failed';
  duration_seconds: number;
  turns_count: number;
  transcript: CallTurn[];
  tools_used: ToolCallEvent[];
  created_at: string;
}

export interface CallStats {
  total_calls: number;
  active_calls: number;
  avg_duration_seconds: number;
  total_addon_queries: number;
}

export interface TaskItem {
  id: string;
  title: string;
  due: string;
  completed: boolean;
  assignee_name: string;
  assignee_avatar: string;
}

export interface ActivityItem {
  id: string;
  type: 'project' | 'lead' | 'proposal' | 'call';
  title: string;
  author: string;
  time_ago: string;
  icon_color: 'blue' | 'green' | 'rose' | 'purple';
}

export interface ChartPoint {
  label: string;
  value: number;
  formatted: string;
  calls: number;
}

export interface FunnelStats {
  new_leads: number;
  contacted: number;
  qualified: number;
  proposal: number;
  won: number;
}

export interface DashboardOverview {
  user_name: string;
  workspace_name: string;
  total_revenue_formatted: string;
  total_revenue_growth: string;
  active_projects_count: number;
  new_projects_count: number;
  tasks_progress_pct: number;
  leads_this_month: number;
  new_leads_count: number;
  chart_period: string;
  chart_points: ChartPoint[];
  funnel: FunnelStats;
  upcoming_tasks: TaskItem[];
  recent_activities: ActivityItem[];
  recent_calls: CallRecord[];
  stats: CallStats;
}

File: backend/app/schemas/call.py
from typing import List, Dict, Any, Optional
from datetime import datetime
from pydantic import BaseModel


class CallRecordResponse(BaseModel):
    id: str
    tenant_id: str
    call_sid: str
    caller: str
    recipient: str
    status: str
    duration_seconds: float
    turns_count: int
    transcript: List[Dict[str, Any]]
    tools_used: List[Dict[str, Any]]
    created_at: datetime
    ended_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class CallStatsResponse(BaseModel):
    total_calls: int
    active_calls: int
    avg_duration_seconds: float
    total_addon_queries: int


class TaskItem(BaseModel):
    id: str
    title: str
    due: str
    completed: bool = False
    assignee_name: str
    assignee_avatar: str


class ActivityItem(BaseModel):
    id: str
    type: str  # 'project', 'lead', 'proposal', 'call'
    title: str
    author: str
    time_ago: str
    icon_color: str  # 'blue', 'green', 'rose', 'purple'


class ChartPoint(BaseModel):
    label: str
    value: float
    formatted: str
    calls: int


class FunnelStats(BaseModel):
    new_leads: int
    contacted: int
    qualified: int
    proposal: int
    won: int


class DashboardOverviewResponse(BaseModel):
    user_name: str = "Riya"
    workspace_name: str = "Apex Media"
    total_revenue_formatted: str
    total_revenue_growth: str
    active_projects_count: int
    new_projects_count: int
    tasks_progress_pct: int
    leads_this_month: int
    new_leads_count: int
    chart_period: str
    chart_points: List[ChartPoint]
    funnel: FunnelStats
    upcoming_tasks: List[TaskItem]
    recent_activities: List[ActivityItem]
    recent_calls: List[CallRecordResponse]
    stats: CallStatsResponse
